packages/amplication-plugin-api/src/plugin/plugin.resolver.ts

```diff
@@ -9,29 +9,46 @@ import { PluginService } from "./plugin.service";
 import { Public } from "../decorators/public.decorator";
 import { GraphQLError } from "graphql";
 import { PluginVersion } from "../pluginVersion/base/PluginVersion";
+import { PluginVersionService } from "../pluginVersion/pluginVersion.service";
+import { ProcessedPluginVersions } from "./plugin.types";
+import { PluginVersionFindManyArgs } from "../pluginVersion/base/PluginVersionFindManyArgs";
 
 @graphql.Resolver(() => Plugin)
 @common.UseGuards(GqlDefaultAuthGuard, gqlACGuard.GqlACGuard)
 export class PluginResolver extends PluginResolverBase {
   constructor(
     protected readonly service: PluginService,
     @nestAccessControl.InjectRolesBuilder()
-    protected readonly rolesBuilder: nestAccessControl.RolesBuilder
+    protected readonly rolesBuilder: nestAccessControl.RolesBuilder,
+    protected readonly pluginVersionService: PluginVersionService
   ) {
     super(service, rolesBuilder);
   }
 
   @Public()
-  @graphql.Query(() => [Plugin], { nullable: true })
-  async githubPlugins(): Promise<Plugin[]> {
+  @graphql.Mutation(() => [Plugin], { nullable: true })
+  async processPluginCatalog(): Promise<ProcessedPluginVersions[]> {
     try {
       const amplicationPlugins = await this.service.githubCatalogPlugins();
       if (
         Object.prototype.toString.call(amplicationPlugins) === "[object String]"
-      )
+      ) {
         throw amplicationPlugins;
+      }
 
-      return amplicationPlugins;
+      const npmPluginsVersions =
+        await this.pluginVersionService.npmPluginsVersions(amplicationPlugins);
+      if (!npmPluginsVersions) throw "Failed to update plugins versions";
```

Comment:
this is tricky, cause you can have a new plugin at Amplication but don't have versions as the user/contributor didn't publish yet. I wouldn't fail this call, but ignore this plugin at client level or at get plugin request

Reply:
I see the possible issue. I'm going to address it.